src/kern/exec.c

```diff
@@ -30,6 +31,20 @@ struct bpf_map_def SEC("maps") output_map = {
 	.max_entries = 1,
 };
 
+struct record_sample {
+	int ppid;
+	int pid;
+	int tgid;
+	char name[PATH_MAX];
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_RINGBUF);
+	__uint(max_entries, 1 << 12);
+} ringbuf SEC(".maps");
+
+long flags = 0;
+
 SEC("lsm/bprm_committed_creds")
 int BPF_PROG(test_void_hook, struct linux_binprm *bprm)
```

Comment:
Why is this called test_void hook and why does it return an int? Please call it exec_audit (in a separate patch) and have the return type as void